examples/test_tsd_bspline.ipynb

```diff
@@ -0,0 +1,462 @@
+    "Dimension = 3\n",
+    "\n",
+    "VecContType = itk.VectorContainer[\n",
+    "    ElementIdentifierType, itk.Point[CoordType, Dimension]\n",
+    "]\n",
+    "bounding_box = itk.BoundingBox[ElementIdentifierType, Dimension, CoordType, VecContType].New()\n",
+    "bounding_box.SetPoints(movingMesh.GetPoints())\n",
+    "bounding_box.ComputeBoundingBox()\n",
+    "\n",
+    "minBounds = np.array(bounding_box.GetMinimum())\n",
+    "maxBounds = np.array(bounding_box.GetMaximum())\n",
+    "\n",
+    "\n",
+    "spacing = np.sqrt(bounding_box.GetDiagonalLength2()) / imageDiagonal\n",
+    "diff = maxBounds - minBounds\n",
+    "\n",
+    "print('Spacing ', spacing)\n",
+    "print('minBounds ', minBounds)\n",
+    "print('maxBounds ', maxBounds)\n",
+    "\n",
+    "fixedImageSize    = [0]*3\n",
+    "fixedImageSize[0] = math.ceil( 1.2 * diff[0] / spacing )\n",
+    "fixedImageSize[1] = math.ceil( 1.2 * diff[1] / spacing )\n",
+    "fixedImageSize[2] = math.ceil( 1.2 * diff[2] / spacing )\n",
+    "\n",
+    "fixedImageOrigin    = [0]*3\n",
+    "fixedImageOrigin[0] = minBounds[0] - 0.1 * diff[0]\n",
+    "fixedImageOrigin[1] = minBounds[1] - 0.1 * diff[1]\n",
+    "fixedImageOrigin[2] = minBounds[2] - 0.1 * diff[2]\n",
+    "\n",
+    "fixedImageSpacing   = np.ones(3)*spacing\n",
+    "fixedImageDirection = np.identity(3)\n",
+    "\n",
+    "\n",
+    "fixedImage = FixedImageType.New()\n",
+    "fixedImage.SetRegions(fixedImageSize)\n",
+    "fixedImage.SetOrigin( fixedImageOrigin )\n",
+    "fixedImage.SetDirection( fixedImageDirection )\n",
+    "fixedImage.SetSpacing( fixedImageSpacing )\n",
+    "fixedImage.Allocate()\n",
+    "\n",
+    "\n",
+    "# Create BSpline Transformation object and initialize the parameters\n",
+    "SplineOrder = 3\n",
+    "TransformType  = itk.BSplineTransform[itk.D, Dimension, SplineOrder]\n",
+    "InitializerType = itk.BSplineTransformInitializer[TransformType, FixedImageType]\n",
+    "\n",
+    "transform = TransformType.New()\n",
+    "\n",
+    "numberOfGridNodesInOneDimension = 5\n",
+    "transformInitializer = InitializerType.New()\n",
+    "transformInitializer.SetTransform(transform)\n",
+    "transformInitializer.SetImage(fixedImage)\n",
+    "transformInitializer.SetTransformDomainMeshSize(numberOfGridNodesInOneDimension - SplineOrder)\n",
+    "transformInitializer.InitializeTransform()\n",
+    "\n",
+    "\n",
+    "\n",
+    "MetricType = itk.ThinShellDemonsMetricv4.MD3\n",
+    "metric = MetricType.New()\n",
+    "metric.SetStretchWeight(stretchWeight)\n",
+    "metric.SetBendWeight(bendWeight)\n",
+    "metric.SetGeometricFeatureWeight(geometricWeight)\n",
+    "metric.UseConfidenceWeightingOn()\n",
+    "metric.UseMaximalDistanceConfidenceSigmaOff()\n",
+    "metric.UpdateFeatureMatchingAtEachIterationOn()\n",
+    "metric.SetMovingTransform(transform)\n",
+    "# Reversed due to using points instead of an image\n",
+    "# to keep semantics the same as in itkThinShellDemonsTest.cxx\n",
```

Comment:
If `itkThinShellDemonsTest.cxx` does something wrong, we should fix it instead of propagate the error.

Reply:
This is what we have been using since the beginning.
There was a comment on this in the README also.

I did try to fix it before but wasn't successful (didn't get the desired result).
I will check about this more.

Reply:
I might have to switch the meshes inside the metric calculation. 
I will test it and report it here. 

You are right, it is better to get it fixed now.